Lib/glyphsLib/builder/glyph.py

```diff
@@ -147,18 +150,23 @@ def to_glyphs_glyph(self, ufo_glyph, ufo_layer, master):
 
     if ufo_glyph.unicodes:
         glyph.unicodes = ["{:04X}".format(c) for c in ufo_glyph.unicodes]
-    note = ufo_glyph.note
-    if note is not None:
-        glyph.note = note
+    glyph.note = ufo_glyph.note or ""
     if GLYPHLIB_PREFIX + "lastChange" in ufo_glyph.lib:
         last_change = ufo_glyph.lib[GLYPHLIB_PREFIX + "lastChange"]
         # We cannot be strict about the dateformat because it's not an official
         # UFO field mentioned in the spec so it could happen to have a timezone
         glyph.lastChange = from_loose_ufo_time(last_change)
     if ufo_glyph.markColor:
         glyph.color = _to_glyphs_color(ufo_glyph.markColor)
+
+    # The export flag can be stored in the glyph's lib key (for upgrading legacy
+    # sources) or the Designspace-level public.skipExportGlyphs lib key (canonical
+    # place to store the information). The UFO level lib key is ignored.
     if GLYPHLIB_PREFIX + "Export" in ufo_glyph.lib:
         glyph.export = ufo_glyph.lib[GLYPHLIB_PREFIX + "Export"]
+    if ufo_glyph.name in self.designspace.lib.get("public.skipExportGlyphs", []):
```

Comment:
maybe the builder should maintain a set of the public.skipExportGlyphs to make this a constant-time lookup instead of linear search in a list.

Reply:
Done. Could be better coordinated with `_fake_designspace` and the other one, we're redoing work, but ugh.